pkgs/build-support/node/fetch-npm-deps/src/main.rs

```diff
@@ -35,10 +35,29 @@ struct OldPackage {
 
 #[derive(Debug, Deserialize, PartialEq, Eq)]
 struct Package {
-    resolved: Option<Url>,
+    resolved: Option<UrlOrString>,
     integrity: Option<String>,
 }
 
+#[derive(Debug, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+enum UrlOrString {
```

Comment:
Nit: I personally like naming this `UrlLike` to emphasize that it's still a URL, just maybe not a valid one.

Reply:
Not sure on this one. I'm talking about URLs like `github:NixOS/nixpkgs` here, which technically *are* valid URLs, but... not.

Maybe yours is best, hm.